kuma/javascript/src/page.jsx

```diff
@@ -0,0 +1,165 @@
+// @flow
+import * as React from 'react';
+import { useContext } from 'react';
+import { css } from '@emotion/core';
+
+import DocumentProvider from './document-provider.jsx';
+import gettext from './gettext.js';
+import { Row, Strut } from './layout.jsx';
+import Header from './header/header.jsx';
+
+const strings = {
+    relatedTopics: gettext('Related Topics')
+};
+
+const styles = {
+    titlebar: css({
+        backgroundColor: '#f5f9fa',
+        padding: '16px 0 16px 32px'
+    }),
+    title: css({
+        fontFamily:
+            'x-locale-heading-primary, zillaslab, "Palatino", "Palatino Linotype", x-locale-heading-secondary, serif',
+        fontSize: '2.8rem',
+        margin: 0
+    }),
+    toc: css({
+        color: '#fff',
+        backgroundColor: '#222',
+        padding: '15px 0 15px 32px',
+        '& li': {
+            display: 'inline-block'
+        },
+        '& a': {
+            color: '#83d0f2',
+            paddingRight: 30,
+            whiteSpace: 'nowrap',
+            textDecoration: 'none',
+            fontSize: '1rem'
+        }
+    }),
+    content: css({
+        alignItems: 'start',
+        padding: '30px 24px'
+    }),
+    article: css({
+        flex: 3,
+        '& p': {
+            maxWidth: '42rem'
+        }
+    }),
+    sidebar: css({
+        flex: 1
+    }),
+    breadcrumbs: css({}),
+    quicklinks: css({})
+};
+
+function Titlebar() {
+    const documentData = useContext(DocumentProvider.context);
+
+    return (
+        documentData && (
+            <Row css={styles.titlebar}>
+                <h1 css={styles.title}>{documentData.title}</h1>
+            </Row>
+        )
+    );
+}
+
+function TOC() {
+    const documentData = useContext(DocumentProvider.context);
+
+    return (
+        documentData && (
+            <Row
+                css={styles.toc}
+                dangerouslySetInnerHTML={{
+                    __html: `<ol>${documentData.tocHTML}</ol>`
+                }}
```

Comment:
This dangerouslySetInnerHTML is how you inject HTML with React. They force you to type "dangerous" like Jinja forces you to type "safe".